code/controllers/subsystem/air.dm

```diff
@@ -68,6 +72,15 @@ SUBSYSTEM_DEF(air)
 /datum/controller/subsystem/air/fire(resumed = 0)
 	var/timer = TICK_USAGE_REAL
 
+	if(pipenets_needing_rebuilt.len) // always do this nonsense first
```

Comment:
length() for speeeeeeeeeeeeeed

Reply:
are you sure, isnt checking iwth a proc slower than checking the var itself

Reply:
Last time I benchmarked the difference wasn't significant neither consistent enough to favor one over another